tests/protocol/test_wire.py

```diff
@@ -815,6 +815,41 @@ def http_get_handler(url, *_, **kwargs):
             self.assertTrue(self.is_host_plugin_extension_artifact_request(urls[3]), "The retry request should have been over the host channel")
             self.assertEquals(HostPluginProtocol.is_default_channel(), False, "The default channel should not have changed")
 
+    def test_upload_logs_should_not_refresh_plugin_when_first_attempt_succeeds(self):
+        def http_put_handler(url, *_, **__):
+            if protocol.get_endpoint() in url and url.endswith('/vmAgentLog'):
+                return MockResponse(body=b'', status_code=200)
+            self.fail('The upload logs request was sent to the wrong url: {0}'.format(url))
```

Comment:
but at this point you do not know that it was an upload log request, you only know that it was a PUT request

Reply:
I was looking at how the test `test_upload_status_blob_should_use_the_host_channel_by_default` was written:
`self.fail('The upload status request was sent to the wrong url: {0}'.format(url))`.

Since this inline put handler is defined in the test method that's (only) calling the upload log (or upload status) method, isn't it safe to assume we are only expecting PUT requests to those URLs?

Reply:
I think I misread your code initially, you are checking is_host_plugin_put_logs_request

as far as the self.fail, let's remove it (see my previous comment). when i initially wrote the mock protocol, it did not raise on unhandled  requests so the fail in test_upload_status_blob_should_use_the_host_channel_by_default must be a left over; you can remove it, too, if you want